assets/src/util/math.ts

```diff
@@ -7,3 +9,46 @@
  */
 export const clamp = (value: number, min: number, max: number): number =>
   Math.min(Math.max(min, value), max)
+
+interface ClosestPosition {
+  position: LatLngExpression
+  index: number
+  distance: number
+}
+
+/**
+ * Finds the element in {@link positions} that is closest to {@link point}
+ * @param positions List of coordinates to search
+ * @param point Point of reference to check distance to
+ * @returns
+ * if {@link positions} is empty, returns `undefined`
+ *
+ * Otherwise, returns a {@link ClosestPosition} object containing
+ * - {@link ClosestPosition.position}: the closest element in {@link positions} to {@link point}
+ * - {@link ClosestPosition.index}: the index of the element in {@link positions}
+ * - {@link ClosestPosition.distance}: the distance to {@link point}
+ */
+export const closestPosition = (
+  positions: LatLngExpression[],
+  point: LatLng
+): ClosestPosition | undefined => {
+  const positionsByDistance = positions
+    .map((currentPosition, index) => ({
+      distance: point.distanceTo(currentPosition),
+      position: currentPosition,
+      index,
+    }))
+    .sort((lhs, rhs) => lhs.distance - rhs.distance)
```

Comment:
I don't believe we have a function like that right now, aside from implementing it as a `reduce` (https://developer.mozilla.org/en-US/search?q=maxby)

I _personally_ found the `reduce` operation harder to read, and this sets us up to do interpolation afterward, and I imagine the `sort` is negligible compared to a single pass `reduce` at the number of points we're working with.

I'd be happy to look into it further if you have any suggestions!

Reply:
Nahh - what you said makes sense. If there are a lot of points, then I could see `sort` being slower, but like you said, marginally - maybe by a factor of 2 or something.

If we start seeing that this is a problem, then we can revisit it - my idea would be to get the two nearest points (which I think could be done in two passes, rather than potentially `log_base_2(n)`).

Reply:
Another benefit, I'm returning `index` because it's possible that a non index adjacent point could be closer than other points nearby

![svg](https://github.com/mbta/skate/assets/6913198/991ec654-9399-4ad7-93bd-89108f6aa8d6)